client/src/main/scala/com.gu.contentapi.client/model/ContentApiError.scala

```diff
@@ -0,0 +1,15 @@
+package com.gu.contentapi.client.model
+
+import com.gu.contentapi.client.model.v1.ErrorResponse
+import com.gu.contentapi.client.thrift.ThriftDeserializer
+import scala.util.Try
+
+case class ContentApiError(httpStatus: Int, httpMessage: String, errorResponse: Option[ErrorResponse] = None) extends Exception(httpMessage)
+
+object ContentApiError {
+  def apply(response: HttpResponse): ContentApiError = {
+    val errorResponse = Try(ThriftDeserializer.deserialize(response.body, ErrorResponse)).toOption
```

Comment:
I wonder whether we should add a check to see if the `response` is an error response...

Reply:
With only an array of bytes, trying to deserialize is the only way to know if it is.